code/modules/mining/mine_items.dm

```diff
@@ -278,153 +278,60 @@
 	icon_state = "capsule"
 	icon = 'icons/obj/mining.dmi'
 	w_class = 1
+	origin_tech = "engineering=3;bluespace=3"
+	var/template_id = "shelter_alpha"
+	var/datum/map_template/shelter/template
 	var/used = FALSE
 
+/obj/item/weapon/survivalcapsule/proc/get_template()
+	if(template)
+		return
+	template = shelter_templates[template_id]
+	if(!template)
+		throw EXCEPTION("Shelter template ([template_id]) not found!")
+		qdel(src)
```

Comment:
This `qdel` will never happen, because `throw`ing immediately exits the current proc. An exception can be non-fatally reported with `log_runtime(exception_goes_here, src)`